src/object/cli_shard.c

```diff
@@ -415,6 +415,10 @@ dc_obj_shard_rw(struct dc_obj_shard *shard, enum obj_rpc_opc opc,
 	orw->orw_dkey_hash = args->dkey_hash;
 	orw->orw_nr = nr;
 	orw->orw_dkey = *dkey;
+	if (args->dkey_csum != NULL)
+		orw->orw_dkey_csum = args->dkey_csum;
+	else
+		memset(&orw->orw_dkey_csum, 0, sizeof(orw->orw_dkey_csum));
```

Comment:
orw->orw_dkey_csum is a pointer right? then orw->orw_dkey_csum = NULL is enough
actually the orw is zeroed by calloc, seems fine to remove the else case.